firmware/projects/LVController/inc/app.h

```diff
@@ -0,0 +1,136 @@
+/// @author Blake Freer
+/// @date 2023-12-25
+
+#include <cmath>
+
+#include "shared/periph/gpio.h"
+#include "shared/periph/pwm.h"
+#include "shared/util/mappers/clamper.h"
+#include "shared/util/mappers/mapper.h"
+
+/**
+ * @brief A Subsystem which can be enabled / disabled.
+ */
+class Subsystem {
+public:
+    /**
+     * @brief Construct a new Subsystem object
+     *
+     * @param enable_output Digital Output which enables the subsystem.
+     * enabled).
+     */
+    Subsystem(shared::periph::DigitalOutput& enable_output)
+        : enable_output_(enable_output) {}
+
+    inline void Enable() const {
+        enable_output_.SetHigh();
+    }
+
+    inline void Disable() const {
+        enable_output_.SetLow();
+    }
+
+private:
+    shared::periph::DigitalOutput& enable_output_;
+};
+
+class Indicator {
+public:
+    Indicator(shared::periph::DigitalOutput& output) : output_(output) {}
+
+    inline void TurnOn() {
+        output_.SetHigh();
+    }
+
+    inline void TurnOff() {
+        output_.SetLow();
+    }
+
+    inline void SetState(bool value) {
+        output_.Set(value);
+    }
+
+private:
+    shared::periph::DigitalOutput& output_;
+};
+
+class Fan : public Subsystem {
+public:
+    Fan(shared::periph::DigitalOutput& enable_output,
+        shared::periph::PWMOutput& pwm_output,
+        shared::util::Mapper<float>& power_to_duty)
+        : Subsystem(enable_output),
+          pwm_output_(pwm_output),
+          power_to_duty_(power_to_duty) {}
+
+    void Enable() {
```

Comment:
```suggestion
    void Enable() override {
```
similar for other overridden methods